kubectl-plugin/pkg/cmd/get/get_workergroup.go

```diff
@@ -0,0 +1,396 @@
+package get
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/ray-project/kuberay/kubectl-plugin/pkg/util"
+	"github.com/ray-project/kuberay/kubectl-plugin/pkg/util/client"
+	"github.com/spf13/cobra"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/cli-runtime/pkg/printers"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+	"k8s.io/kubectl/pkg/util/templates"
+
+	rayv1 "github.com/ray-project/kuberay/ray-operator/apis/ray/v1"
+)
+
+type GetWorkerGroupsOptions struct {
+	configFlags   *genericclioptions.ConfigFlags
+	ioStreams     *genericclioptions.IOStreams
+	namespace     string
+	cluster       string
+	workerGroup   string
+	allNamespaces bool
+}
+
+type enrichedWorkerGroupSpec struct {
+	namespace string
+	cluster   string
+	spec      rayv1.WorkerGroupSpec
+}
+
+type workerGroup struct {
+	totalCPU        resource.Quantity
+	totalGPU        resource.Quantity
+	totalTPU        resource.Quantity
+	totalMemory     resource.Quantity
+	namespace       string
+	name            string
+	cluster         string
+	readyReplicas   int32
+	desiredReplicas int32
+}
+
+var getWorkerGroupsExample = templates.Examples(`
+		# Get worker groups in the default namespace
+		kubectl ray get workergroup
+
+		# Get worker groups in all namespaces
+		kubectl ray get workergroup --all-namespaces
+
+		# Get all worker groups in a namespace
+		kubectl ray get workergroup --namespace my-namespace
+
+		# Get all worker groups for Ray clusters named my-raycluster in all namespaces
+		kubectl ray get workergroup --ray-cluster my-raycluster --all-namespaces
+
+		# Get all worker groups in a namespace for a Ray cluster
+		kubectl ray get workergroup --namespace my-namespace --ray-cluster my-raycluster
+
+		# Get one worker group in a namespace for a Ray cluster
+		kubectl ray get workergroup my-group --namespace my-namespace --ray-cluster my-raycluster
+	`)
+
+func NewGetWorkerGroupOptions(streams genericclioptions.IOStreams) *GetWorkerGroupsOptions {
+	return &GetWorkerGroupsOptions{
+		configFlags: genericclioptions.NewConfigFlags(true),
+		ioStreams:   &streams,
+	}
+}
+
+func NewGetWorkerGroupCommand(streams genericclioptions.IOStreams) *cobra.Command {
+	options := NewGetWorkerGroupOptions(streams)
+	cmdFactory := cmdutil.NewFactory(options.configFlags)
+
+	cmd := &cobra.Command{
+		Use:          "workergroup [GROUP] [(-c/--ray-cluster) RAYCLUSTER]",
+		Aliases:      []string{"workergroups"},
+		Short:        "Get Ray worker groups",
+		Example:      getWorkerGroupsExample,
+		SilenceUsage: true,
+		Args:         cobra.MaximumNArgs(1),
+		// ValidArgsFunction: completion.RayClusterCompletionFunc(cmdFactory),
```

Comment:
nit: Remove this line or write a new shell completion funciton.

Reply:
removed

Reply:
issue to track https://github.com/ray-project/kuberay/issues/3051